include/deal.II/lac/packaged_operation.h

```diff
@@ -829,6 +829,208 @@ operator*(const PackagedOperation<Range> &comp,
+      Range_2 &tmp_g1 = *(vector_memory_g.alloc());
+      Range_2 &tmp_g2 = *(vector_memory_g.alloc());
+
+      // Reinitialise in context of how they'll be used
+      A_inv.reinit_range_vector(tmp_f1, /*bool omit_zeroing_entries =*/ true);
+      C.reinit_range_vector(tmp_g1, /*bool omit_zeroing_entries =*/ true);
+
+      // Condensation on RHS of one field
+      // Need to form g* such that g* = g - C*A_inv*f
+      try
+        {
+          A_inv.vmult(tmp_f1, f); // tmp_f1 = A_inv * f
+        }
+      catch (...)
+        {
+          AssertThrow(false,
+                      ExcMessage("No convergence in A_inv vmult operation"));
+        }
+      C.vmult(tmp_g1, tmp_f1); // tmp2 = C * A_inv * f
+
+      g_star = g;
+      g_star -= tmp_g1; // tmp_g2 = g - C * A_inv * f
+    };
+
+    return_comp.apply_add = [&A_inv, &C, &f, &g](Range_2 &g_star)
+    {
+
+      static GrowingVectorMemory<Range_1> vector_memory_f;
+      static GrowingVectorMemory<Range_2> vector_memory_g;
+
+      Range_1 &tmp_f1 = *(vector_memory_f.alloc());
+      Range_2 &tmp_g1 = *(vector_memory_g.alloc());
+      Range_2 &tmp_g2 = *(vector_memory_g.alloc());
+
+      // Reinitialise in context of how they'll be used
+      A_inv.reinit_range_vector(tmp_f1, /*bool omit_zeroing_entries =*/ true);
+      C.reinit_range_vector(tmp_g1, /*bool omit_zeroing_entries =*/ true);
+
+      // Condensation on RHS of one field
+      // Need to form g* such that g* = g - C*A_inv*f
+      try
+        {
+          A_inv.vmult(tmp_f1, f); // tmp_f1 = A_inv * f
+        }
+      catch (...)
+        {
+          AssertThrow(false,
+                      ExcMessage("No convergence in A_inv vmult operation"));
+        }
+      C.vmult(tmp_g1, tmp_f1); // tmp2 = C * A_inv * f
+
+      g_star += g;
+      g_star -= tmp_g1; // tmp_g2 = g - C * A_inv * f
+    };
+
+    return return_comp;
+  }
+
+  /**
+   * @relates PackagedOperation
+   *
+   * For the system of equations
+   * @f{eqnarray*}
+       (1) \quad Ax + By &=& f \\
+       (2) \quad Cx + Dy &=& g \quad ,
+   * @f}
+   * this operation performs the post-processing step of the
+   * Schur complement to solve for the second subvector @p x once
+   * subvector @p y is known:
+   * @f[
+       x =  A^{-1}(f - By)
+     @f]
+   *
+   * @see
+   * @ref GlossBlockLA "Block (linear algebra)"
+   * @author Jean-Paul Pelteret, 2015
+   *
+   * @ingroup LAOperators
+   */
+  template <typename Range_1, typename Domain_1,
+            typename Range_2, typename Domain_2>
+  PackagedOperation<Domain_1>
+  postprocess (const LinearOperator<Range_1, Domain_1> &A_inv,
+               const LinearOperator<Range_1, Domain_2> &B,
+               const Domain_2                          &y,
+               const Range_1                           &f)
+  {
+    PackagedOperation<Domain_1> return_comp;
+
+    return_comp.reinit_vector = A_inv.reinit_domain_vector;
+
+    return_comp.apply = [&A_inv, &B, &y, &f](Domain_1 &x)
```

Comment:
Maybe we can avoid the code duplication by putting the ```apply_add``` first and catching it in ```apply```? :-)
```
const auto apply_add = return_comp.apply_add;
return_comp.apply = [apply_add](Range &v)
{
  v = 0.;
  apply_add(v);
};
```